src/ScriptBuilder/Writers/OutboxWriter.cs

```diff
@@ -1,21 +1,41 @@
+using System;
 using System.IO;
 using NServiceBus.Persistence.Sql.ScriptBuilder;
 
 class OutboxWriter
 {
-    public static void WriteOutboxScript(string scriptPath, BuildSqlDialect sqlDialect)
+    public static void WriteOutboxScript(bool clean, bool overwrite, string scriptPath, BuildSqlDialect sqlDialect)
     {
         var createPath = Path.Combine(scriptPath, "Outbox_Create.sql");
-        File.Delete(createPath);
+        if (clean)
```

Comment:
There's a lot of duplicated logic for each file and for each type of script we output. I wonder if it's worth pulling some of it into a base class and then having the outbox writer look something like this:

```csharp
class OutboxWriter : ScriptWriter
{
  ctor(clean, overwrite, scriptPath) : base(clean, overwrite, scriptPath) { ... }
  
  public void WriteOutboxScript(BuildSqlDialect sqlDialect)
  {
    WriteScript("Outbox_Create.sql", writer => OutboxScriptBuilder.BuildCreateScript(writer, sqlDialect));
    WriteScript("Outbox_Drop.sql", writer => OutboxScriptBuilder.BuildDropScript(writer, sqlDialect));
  }
}
```

Reply:
done